coderd/database/queries/notificationsinbox.sql

```diff
@@ -0,0 +1,38 @@
+-- name: FetchUnreadInboxNotificationsByUserID :many
+SELECT * FROM inbox_notifications WHERE user_id = $1 AND read_at IS NULL ORDER BY created_at DESC;
+
+-- name: FetchInboxNotificationsByUserID :many
+SELECT * FROM inbox_notifications WHERE user_id = $1 ORDER BY created_at DESC;
```

Comment:
These queries are identical except for `read_at IS NULL`; rather parameterise `FetchInboxNotificationsByUserID` so you can explicitly filter for read/unread, or neither.

`FetchUnreadInboxNotificationsByUserIDFilteredByTemplatesAndTargets` could be refactored out according to the same pattern, too.

Reply:
As I tried to iterate here and from what I found on the internet - I can't find a way to achieve that with sqlc.

sqlc, with narg, allow to have null parameters and optional values, with default values if we want - here the objective is a bit different as we want to completely change the query and the `WHERE` clause based on if a parameter is present or not.

If you have more idea, I could check.